src/client/eppo-client-assignment-details.spec.ts

```diff
@@ -220,32 +248,57 @@ describe('EppoClient get*AssignmentDetails', () => {
     afterAll(() => {
       jest.restoreAllMocks();
     });
-    it.each(readAssignmentTestData())(
-      'test variation assignment details',
-      async ({ flag, variationType, defaultValue, subjects }: IAssignmentTestCase) => {
-        const client = new EppoClient(storage);
-        client.setIsGracefulFailureMode(false);
-
-        const typeAssignmentDetailsFunctions = {
-          [VariationType.BOOLEAN]: client.getBooleanAssignmentDetails.bind(client),
-          [VariationType.NUMERIC]: client.getNumericAssignmentDetails.bind(client),
-          [VariationType.INTEGER]: client.getIntegerAssignmentDetails.bind(client),
-          [VariationType.STRING]: client.getStringAssignmentDetails.bind(client),
-          [VariationType.JSON]: client.getJSONAssignmentDetails.bind(client),
-        };
-
-        const assignmentFn = typeAssignmentDetailsFunctions[variationType];
-        if (!assignmentFn) {
-          throw new Error(`Unknown variation type: ${variationType}`);
-        }
-
-        const assignments = getTestAssignmentDetails(
-          { flag, variationType, defaultValue, subjects },
-          assignmentFn,
-        );
-
-        validateTestAssignmentDetails(assignments, flag);
-      },
-    );
+
+    describe.each(getTestFilePaths())('for file: %s', (testFilePath: string) => {
+      const testCase = parseJSON(testFilePath);
+      describe.each(testCase.subjects.map(({ subjectKey }) => subjectKey))(
```

Comment:
I re-wrote how this works so that the error messages are more clear and so that it's easier to debug